### PR TITLE
Useful Icarus Drones

### PR DESCRIPTION
**Pull request by Geevies**
* The NMV Icarus will now dispatch combat drones to assist with medium to large carp migrations.
* Interacting with simple mobs now have full stops in the chat menu.
* For the admins, combat drones and cavern dwellers fighting eachother will no longer make admin messages, unless a human player is hit. In that case, laugh at them.
* Malfunctioning Icarus drones now have extended descriptions that tell you a little about what they're doing.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/mob/living/simple_animal/hostile/retaliate/icarus_drone.dm [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+		return TRUE
+	if(istype(L, /mob/living/simple_animal/hostile/retaliate/cavern_dweller))
+		return TRUE
+	if(istype(L, /mob/living/simple_animal/hostile/giant_spider))
+		return TRUE
+	if(istype(L, /mob/living/simple_animal/hostile/spider_queen))
+		return TRUE
+	if(istype(L, /mob/living/simple_animal/hostile/retaliate/malf_drone))
+		return TRUE
+	return FALSE
```
Comment (Doxxmedearly) [inline code/modules/mob/living/simple_animal/hostile/retaliate/icarus_drone.dm]: You should probably toss all these mobs in a list and check is_type_in_list(L, listname).
That might actually be a good thing to slap on the mob/blah/hostile parent for fine-tuning future mob targeting? And set the drone's list to all these things. If you feel like it.

Comment (Geevies) [reply code/modules/mob/living/simple_animal/hostile/retaliate/icarus_drone.dm]: alright

File: code/modules/mob/living/simple_animal/hostile/retaliate/icarus_drone.dm [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+	projectiletype = /obj/item/projectile/beam/drone
+	projectilesound = 'sound/weapons/laser3.ogg'
+	destroy_surroundings = FALSE
+	var/datum/effect_system/ion_trail/ion_trail
+
+	var/turf/patrol_target
+	var/explode_chance = 1
+	var/disabled = FALSE
+	var/exploding = FALSE
+	var/list/valid_living_targets = list(
```
Comment (Geevies) [inline code/modules/mob/living/simple_animal/hostile/retaliate/icarus_drone.dm]:
```suggestion
	valid_living_targets = list(
```

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 3]
```diff
@@ -350,22 +353,30 @@ mob/living/simple_animal/hostile/hitby(atom/movable/AM as mob|obj,var/speed = TH
...
+
+	if(length(valid_living_targets)) // it has a specific list of targets
+		if(is_type_in_list(L, valid_living_targets))
 			return TRUE
+		return FALSE
+	else
+		if(ismech(L))
+			var/mob/living/heavy_vehicle/M = L
+			if(!M.pilots?.len)
+				return FALSE
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: Why did you pull this all the way down? Having it at the top makes it skip some calculations, which is a good idea for AI.

File: code/modules/mob/living/simple_animal/hostile/retaliate/icarus_drone.dm [hunk 4]
```diff
@@ -0,0 +1,263 @@
+// friendly icarus combat drones
+/mob/living/simple_animal/hostile/retaliate/icarus_drone
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/hostile/retaliate/icarus_drone.dm]: I'm not sure if I agree with the amount of copypaste here. A lot of this shit is already defined on malf drones. I'd say generalize the drone and then split it.

**Comment by Geevies** [comment]
separating the stuff is too much effort for what i get out of this PR right now

**Pull request closed without merging**